packages/tpetra/core/src/Tpetra_Details_packCrsMatrix_def.hpp

```diff
@@ -1068,15 +1094,62 @@ packCrsMatrixWithOwningPIDs (const CrsMatrix<ST, LO, GO, NT>& sourceMatrix,
                                             sourcePIDs.size (), true,
                                             "export_pids");
   constexpr bool pack_pids = true;
-  PackCrsMatrixImpl::packCrsMatrix(
-      sourceMatrix, exports_dv, num_packets_per_lid_d, export_lids_d,
-      export_pids_d, constantNumPackets, pack_pids, distor);
+  try {
```

Comment:
@mhoemmen, is a try/catch really needed here?  Is the function `packCrsMatrixWithOwningPIDs()` not allowed to throw?  It seems like this might be a poor usage of exception handling (but I would have to know more about this code).

Reply:
@bartlettroscoe This is the debug-mode branch of the code.  Normal release-mode execution doesn't take this code path.  (Tpetra's "debug mode" is enabled by default in a debug build and disabled by default in a release build, but users can control this at run time with the `TPETRA_DEBUG` environment variable.)

The issue is with `Teuchos::ArrayView` throwing ;-) and it's really hard to find all possible uses of such a fundamental class.  I'm just trying to be brutally conservative so that I can catch potential hangs in the code.  

In general, if a function like `packCrsMatrix` throws, the code would probably hang in an MPI run with multiple processes.

Reply:
@mhoemmen, if this is a common idiom in Tpetra, then perhpas should be a macro that encapsulates this and removes a lot of this duplication and boilerplate code.   You can put hooks into macros calls that allow stopping a debugger.  Just look at `TEUCHOS_TEST_FOR_EXCEPTION()` functions.